src/plugins/intel_gpu/src/graph/impls/cpu/read_value.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "read_value_inst.h"
+#include "impls/implementation_map.hpp"
+#include "register.hpp"
+
+namespace cldnn {
+namespace cpu {
+
+struct read_value_impl : public typed_primitive_impl<read_value> {
+    std::unique_ptr<primitive_impl> clone() const override {
+        return make_unique<read_value_impl>(*this);
+    }
+
+    event::ptr execute_impl(const std::vector<event::ptr>& events, read_value_inst& instance) override {
+        for (auto e : events) {
+            e->wait();
+        }
+        const auto arg = instance.argument;
+        const auto variable_id = arg.variable_id;
+
+        auto& variable = instance.get_network().get_memory_state(variable_id);
+
+        if (variable.memory->get_layout() != arg.output_layout) {
+            CLDNN_ERROR_MESSAGE(instance.id(), "Layout mismatch");
+        }
+
+        if (!variable.is_set) {
+            auto &stream = instance.get_network().get_stream();
+            const auto ev_set_output = instance.output_memory().fill(stream, 0);
+            return ev_set_output;
+        }
```

Comment:
Since the memory is allocated with reset=true mode, I think we dont need to fill it with 0 again. 
So I am curious whether we need this MemoryState structure just to hold the "is_set"?

Reply:
If the reason that we need that "MemoryState" structure is to hold the "is_set" and the behavior of the read_value is returning 0 if "is_set = false", then I think we dont need that structure because the allocated memory with (reset=true) will 0 as initial value already.

Reply:
In the latest patchset I added [InferRequest::QueryState() API implementation](https://docs.openvino.ai/2021.2/openvino_docs_IE_DG_network_state_intro.html#openvino_state_api) which depends on `is_set` flag logic.